### PR TITLE
fix: Test using the object store wrapper interface

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #863, the "test" part. I'm still experimenting with the "rework" part :)

Also fixing the tests caught another `unimplemented` left in there... 😳

### CODE REVIEW

File: object_store/src/aws.rs [hunk 1]
```diff
@@ -520,25 +522,30 @@ mod tests {
...
         config.region = "us-west-1".into();
 
-        let integration = AmazonS3::new(
-            config.access_key_id,
-            config.secret_access_key,
-            config.region,
-            &config.bucket,
-        )
-        .expect("Valid S3 config");
+        let integration = ObjectStore::new_amazon_s3(
```
Comment (alamb) [inline object_store/src/aws.rs:525]: 👍

File: object_store/src/lib.rs [hunk 2]
```diff
@@ -300,7 +300,11 @@ impl ObjectStoreApi for ObjectStore {
                 .map_ok(|list_result| list_result.map_paths(path::Path::File))
                 .await
                 .context(FileObjectStoreError),
-            (MicrosoftAzure(_azure), _) => unimplemented!(),
```
Comment (alamb) [inline object_store/src/lib.rs:303]: 🎉  Testing for the Win

**Review by alamb — APPROVED** [review]
Testing for the win -- looks like a real improvement to me @carols10cents . Thank you@

**Review by mkmik — APPROVED** [review]
LGTM

**Review by domodwyer — APPROVED** [review]

**Pull request merged**